src/tools/schema2kotlin.ts

```diff
@@ -55,33 +55,82 @@ ${withCustomPackage(`import arcs.Particle;
 import arcs.Entity
 import arcs.StringEncoder
 import arcs.StringDecoder
-`)}`;
+`)}
+import kotlin.reflect.KProperty\n`;
   }
 
   getClassGenerator(node: SchemaNode): ClassGenerator {
     return new KotlinGenerator(node);
   }
+
+  fileFooter(): string {
+    console.log(`child`);
+    return `
+class TextDelegate {
+    var isSet = false
+    var v = ""
+    operator fun getValue(thisRef: Any?, property: KProperty<*>): String {
+        return v
+    }
+
+    operator fun setValue(thisRef: Any?, property: KProperty<*>, value: String) {
+        this.isSet = true
+        this.v = value
+    }
+}
+
+class NumDelegate {
+    var isSet = false
+    var v = 0.0
+    operator fun getValue(thisRef: Any?, property: KProperty<*>): Double {
+        return v
+    }
+
+    operator fun setValue(thisRef: Any?, property: KProperty<*>, value: Double) {
+        this.isSet = true
+        this.v = value
+    }
+}
+
+class BooleanDelegate {
+    var isSet = false
+    var v = false
```

Comment:
These variables (also the ones in NumDelegate and TextDelegate) should probably be `private`

Reply:
👍